### ISSUE TITLE
UnboundLocalError: local variable 'inv_port' referenced before assignment

### ISSUE THREAD

**Issue by msipos** [ansible/ansible#14654]
##### Issue Type:
- Bug Report
##### Ansible Version:

```
ansible 2.0.1.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```
##### Ansible Configuration:

No configuration
##### Environment:

Running from Linux Mint, managing Ubuntu 14.04.
##### Summary:

When running `syncronize` task, task fails with error:

```
UnboundLocalError: local variable 'inv_port' referenced before assignment

fatal: [...]: FAILED! => {"failed": true, "msg": "Unexpected failure during module execution.", "stdout": ""}
```
##### Steps To Reproduce:

Here is the task I used:

```
- name: ...
  synchronize: src=/home/... dest=/var/... checksum=yes archive=no recursive=yes delete=yes dest_port=5555
  become_user: myuser
```
##### Expected Results:

Syncronization step worked until update to 2.0.1.0
##### Actual Results:

```
An exception occurred during task execution. The full traceback is:
Traceback (most recent call last):
  File "/usr/lib/pymodules/python2.7/ansible/executor/task_executor.py", line 120, in run
    res = self._execute()
  File "/usr/lib/pymodules/python2.7/ansible/executor/task_executor.py", line 418, in _execute
    result = self._handler.run(task_vars=variables)
  File "/usr/lib/pymodules/python2.7/ansible/plugins/action/synchronize.py", line 289, in run
    dest = self._process_remote(dest_host, dest, user, inv_port in localhost_ports)
UnboundLocalError: local variable 'inv_port' referenced before assignment
```

**Comment by msipos** [comment]
To add, this "broke" after upgrading from `2.0.0.2-1ppa~trusty` to `2.0.1.0-1ppa~trusty`.

**Comment by msipos** [comment]
I believe the bug is introduced in https://github.com/ansible/ansible/commit/cb2b19ced8b93b6688370f895c2183ecbd98b3ae by @abadger .

`inv_port` variable is not set in case `dest_port` task argument is set.

In this case I removed `dest_port` argument as it is the same as my regular SSH port.  I added it in in the first place because it didn't used to work with non-22 ports before.

**Comment by abadger** [comment]
Thanks for reporting this.  Fixed on devel and pushed to stable-2.0 branch (might miss 2.0.2 but will be there if we make 2.0.3).
